rethinkdb/datadog_checks/rethinkdb/document_db/query.py

```diff
@@ -0,0 +1,118 @@
+        self.enumerations = [] if enumerations is None else enumerations
+        self.groups = [] if groups is None else groups
+
+    def _make_metric_from_spec(self, document, spec, tags, logger):
+        # type: (Any, MetricSpec, List[str], CheckLoggingAdapter) -> Metric
+        logger.trace('make_metric_from_spec %r', spec)
+
+        path = spec['path']
+        name = spec.get('name', path)
+        value = lookup_dotted(document, path=path)
+
+        if 'modifier' in spec and spec['modifier'] is not None:
+            value = self._modify(value, modifier=spec['modifier'], logger=logger)
+
+        if not isinstance(value, (int, float)):  # pragma: no cover
+            raise RuntimeError('Expected float or int, got {!r} of type {}', value, type(value))
+
+        name = dotted_join(('rethinkdb', self.prefix, name))
+
+        return {'type': spec['type'], 'name': name, 'value': value, 'tags': tags}
+
+    def _make_metrics_from_enumeration(self, document, enumeration, tags, logger):
+        # type: (Any, Enumeration, List[str], CheckLoggingAdapter) -> Iterator[Metric]
+        logger.trace('make_metrics_from_enumeration enumeration=%r', enumeration)
+
+        values = lookup_dotted(document, path=enumeration['path'])  # type: Sequence
+
+        for index, value in enumerate(values):
+            item_tags = tags + ['{}:{}'.format(enumeration['index_tag'], index)]
+
+            for spec in enumeration['metrics']:
+                spec = {
+                    'type': spec['type'],
+                    'name': dotted_join((enumeration['path'], spec['path']), drop_empty=True),
+                    'path': spec['path'],
+                    'modifier': spec.get('modifier'),
+                }
+                yield self._make_metric_from_spec(value, spec, tags=item_tags, logger=logger)
+
+    def _make_metrics_from_group(self, document, group, tags, logger):
+        # type: (Any, Group, List[str], CheckLoggingAdapter) -> Iterator[Metric]
+        logger.trace('make_metrics_from_group group=%r', group)
+
+        mapping = lookup_dotted(document, path=group['path'])  # type: Mapping
+
+        for key in mapping:
+            item_tags = tags + ['{}:{}'.format(group['key_tag'], key)]
+            spec = {
+                'type': group['value_metric_type'],
+                'name': group['path'],
+                'path': key,
+            }  # type: MetricSpec
+            yield self._make_metric_from_spec(mapping, spec, tags=item_tags, logger=logger)
+
+    def _modify(self, value, modifier, logger):
+        # type: (Any, Modifier, CheckLoggingAdapter) -> float
+        logger.trace('modify value=%r modifier=%r', value, modifier)
+
+        if modifier == 'total':
+            return len(value)
+
+        if modifier == 'ok_warning':
+            return AgentCheck.OK if value else AgentCheck.WARNING
+
+        if modifier == 'timestamp':
+            return to_timestamp(value)
```

Comment:
This is mostly useless imo, users would have easier time using a timedelta like https://github.com/DataDog/integrations-core/pull/5524

Reply:
Good point, converted to submit an elapsed time in seconds, LMK if you notice anything weird wrt timezone handling. :-)